app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java

```diff
@@ -131,4 +192,119 @@ private void expandSelected(boolean expand) {
+    }
+
+    /**
+     * On change username button click
+     */
+    public void changeUsernameButton(View view) {
+        // Hide username
+        findViewById(R.id.profile_username).setVisibility(View.GONE);
+
+        // Hide change button
+        findViewById(R.id.profile_change_username).setVisibility(View.GONE);
+
+        // Show edit text
+        EditText changeUsername = findViewById(R.id.profile_username_edit);
+        changeUsername.setVisibility(View.VISIBLE);
+        changeUsername.requestFocus();
+
+        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+        imm.showSoftInput(changeUsername, InputMethodManager.SHOW_IMPLICIT);
+
+        changeUsername.setOnKeyListener((v, keyCode, event) -> {
+            if ((event.getAction() == KeyEvent.ACTION_DOWN) &&
+                    (keyCode == KeyEvent.KEYCODE_ENTER)) {
+                confirmUsernameButton();
+                return true;
+            }
+            return false;
+        });
+    }
+
+    /**
+     * On confirm username change
+     */
+    public void confirmUsernameButton() {
+        EditText usernameEdit = ((EditText)findViewById(R.id.profile_username_edit));
+        hideKeyboard();
+        String newUsername = usernameEdit.getText().toString();
+        usernameEdit.getText().clear();
+
+        // Start a new thread that will handle the process
+        Thread changeThread = new Thread(){
+            @Override
+            public void run() {
+                // Change the username and wait for the task to end
+                ProfileOutcome result = AuthService.getInstance().getAuthAccount().changeUsername(newUsername);
+
+                // Update the view on the UI thread
+                runOnUiThread(() -> {
+                    ((EditText)findViewById(R.id.profile_username_edit)).getText().clear();
+
+                    // If username has changed, update the username text view
+                    if(result == ProfileOutcome.USERNAME_CHANGED || result == ProfileOutcome.USERNAME_REGISTERED) ((TextView)findViewById(R.id.profile_username)).setText(AuthService.getInstance().getAuthAccount().getUsername());
+                    if(result == ProfileOutcome.USERNAME_REGISTERED) fillListView();
+
+                    // Display the message
+                    Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), result.getMessage(), Snackbar.LENGTH_LONG);
+                    snackbar.show();
+
+                    // Hide username edit
+                    findViewById(R.id.profile_username_edit).setVisibility(View.GONE);
+
+                    // Show username
+                    findViewById(R.id.profile_username).setVisibility(View.VISIBLE);
+
+                    // Show change button
+                    findViewById(R.id.profile_change_username).setVisibility(View.VISIBLE);
+                });
+            }
+        };
+        changeThread.start();
+    }
+
```

Comment:
Add method to update UI after that the user changed or cancelled his username change
```suggestion
    /**
     * Hides the keyboard and updates the UI after the username was changed or cancelled
     */
    private void setUiUsernameChanged() {
        // Hide username edit
        findViewById(R.id.profile_username_edit).setVisibility(View.GONE);

        // Show username
        findViewById(R.id.profile_username).setVisibility(View.VISIBLE);

        // Show change button
        findViewById(R.id.profile_change_username).setVisibility(View.VISIBLE);
    }
```